internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -400,6 +400,13 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			// storage_mb size...
 			storageTiers := storageTierMappings[newMb]
 
+			// set default tier for storage when there is now configuration available
+			if v := diff.GetRawConfig().AsValueMap()["storage_tier"]; v.IsNull() {
+				newTier = string(storageTiers.DefaultTier)
+				// set the new value that the update function will pickup and deploy new new storage tier.
+				diff.SetNew("storage_tier", newTier)
+			}
```

Comment:
I think we can remove all of this code and handle everything in the `!isValid` check below:

```suggestion
```